src/query/pipeline/transforms/src/processors/transforms/transform_block_compact.rs

```diff
@@ -43,6 +43,46 @@ impl Compactor for BlockCompactor {
         "BlockCompactTransform"
     }
 
+    fn use_partial_compact() -> bool {
+        true
+    }
+
+    fn compact_partial(&self, blocks: &mut Vec<DataBlock>) -> Result<Vec<DataBlock>> {
+        if blocks.is_empty() {
+            return Ok(vec![]);
+        }
+
+        let size = blocks.len();
+        let mut res = Vec::with_capacity(size);
+        let block = blocks[size - 1].clone();
```

Comment:
why only check the last block? assuming the block in this  `input_state` with the similar size?

Reply:
It will check when one block is pushed into the blocks. So the last block is the newest block